test/tests/selftest/pbs_dshutils_tests.py

```diff
@@ -86,3 +89,38 @@ def test_pbs_env(self):
         self.du.unset_pbs_environment(environ=['pbs_foo'])
         environ = self.du.parse_pbs_environment()
         self.assertNotIn('pbs_foo', environ, msg)
+
+    def check_access(self, path, mode=0o700, user=None, group=None):
+        """
+        Helper function to check user, group and mode of given path
+        """
+        result = os.stat(path)
+        self.assertEqual(oct(result.st_mode & 0o777), oct(mode))
+        if user is not None:
+            uid = pwd.getpwnam(str(user)).pw_uid
+            self.assertEqual(result.st_uid, uid)
+        if group is not None:
+            gid = grp.getgrnam(str(group))[2]
+            self.assertEqual(result.st_gid, gid)
+
+    def test_create_dir_as_user(self):
+        """
+        Test creating a directory as user
+        """
+
+        # check default configurations
+        tmpdir = self.du.create_temp_dir()
+        self.check_access(tmpdir, user=getpass.getuser())
+
+        # create a directory as a specific user
+        tmpdir = self.du.create_temp_dir(asuser=TEST_USER2)
+        self.check_access(tmpdir, user=TEST_USER2, mode=0o755)
+
+        # create a directory as a specific user and permissions
+        tmpdir = self.du.create_temp_dir(asuser=TEST_USER2, mode=0o765)
+        self.check_access(tmpdir, mode=0o765, user=TEST_USER2)
+
```

Comment:
You're creating a directory, I wouldn't choose a mask without the execute bit.  Maybe 750?